ldaptor/test/test_ldapclient.py

```diff
@@ -203,12 +203,16 @@ def collect_result_(result):
         )
         resp_bytestring = response.toWire()
         client.dataReceived(resp_bytestring)
+        self.assertEqual(response.value, results[0])
+        self.assertFalse(d.called)
```

Comment:
Making sure the deferred is not fired until the multi-response handler returns `True`